File: app/src/main/java/com/OdiousPanda/rainbow/DataModel/Unsplash/Unsplash.java
package com.OdiousPanda.rainbow.DataModel.Unsplash;

import java.util.List;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Unsplash {
    @SerializedName("id")
    @Expose
    public String id;
    @SerializedName("created_at")
    @Expose
    public String createdAt;
    @SerializedName("updated_at")
    @Expose
    public String updatedAt;
    @SerializedName("width")
    @Expose
    public Integer width;
    @SerializedName("height")
    @Expose
    public Integer height;
    @SerializedName("color")
    @Expose
    public String color;
    @SerializedName("description")
    @Expose
    public String description;
    @SerializedName("alt_description")
    @Expose
    public String altDescription;
    @SerializedName("urls")
    @Expose
    public Urls urls;
    @SerializedName("links")
    @Expose
    public Links links;
    @SerializedName("categories")
    @Expose
    public List<Object> categories = null;
    @SerializedName("sponsored")
    @Expose
    public Boolean sponsored;
    @SerializedName("sponsored_by")
    @Expose
    public Object sponsoredBy;
    @SerializedName("sponsored_impressions_id")
    @Expose
    public Object sponsoredImpressionsId;
    @SerializedName("likes")
    @Expose
    public Integer likes;
    @SerializedName("liked_by_user")
    @Expose
    public Boolean likedByUser;
    @SerializedName("current_user_collections")
    @Expose
    public List<Object> currentUserCollections = null;
    @SerializedName("user")
    @Expose
    public User user;
    @SerializedName("exif")
    @Expose
    public Exif exif;
    @SerializedName("views")
    @Expose
    public Integer views;
    @SerializedName("downloads")
    @Expose
    public Integer downloads;

}

File: app/src/main/java/com/odiousPanda/rainbowKt/model/dataSource/unsplash/Unsplash.kt
package com.odiousPanda.rainbowKt.model.dataSource.unsplash

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Unsplash {
    @SerializedName("id")
    @Expose
    var id: String = ""

    @SerializedName("created_at")
    @Expose
    var createdAt: String = ""

    @SerializedName("updated_at")
    @Expose
    var updatedAt: String = ""

    @SerializedName("width")
    @Expose
    var width: Int = 0

    @SerializedName("height")
    @Expose
    var height: Int = 0

    @SerializedName("color")
    @Expose
    var color: String = ""

    @SerializedName("description")
    @Expose
    var description: String? = null

    @SerializedName("alt_description")
    @Expose
    var altDescription: String = ""

    @SerializedName("urls")
    @Expose
    var urls: Urls = Urls()

    @SerializedName("links")
    @Expose
    var links: Links = Links()

    @SerializedName("categories")
    @Expose
    var categories: List<Any>? = null

    @SerializedName("sponsored")
    @Expose
    var sponsored: Boolean = false

    @SerializedName("sponsored_by")
    @Expose
    var sponsoredBy: String = ""

    @SerializedName("sponsored_impressions_id")
    @Expose
    var sponsoredImpressionsId: String = ""

    @SerializedName("likes")
    @Expose
    var likes: Int = 0

    @SerializedName("liked_by_user")
    @Expose
    var likedByUser: Boolean = false

    @SerializedName("current_user_collections")
    @Expose
    var currentUserCollections: List<Any>? = null

    @SerializedName("user")
    @Expose
    var user: User = User()

    @SerializedName("exif")
    @Expose
    var exif: Exif = Exif()

    @SerializedName("views")
    @Expose
    var views: Int = 0

    @SerializedName("downloads")
    @Expose
    var downloads: Int = 0
}
